### PR TITLE
Feature shrimp app eli part 2

### PR DESCRIPTION
**Pull request by EliNovakova**

### CODE REVIEW

File: src/shrimps/views.py [hunk 1]
```diff
@@ -1,3 +1,44 @@
...
+    num_shrimps = Shrimp.objects.count()
+    # num_shrimps = len(Shrimp.objects.all())
+
+    # count() - returns an int representing the number of objects in database
+    # is better option if all we want to do it to get the count and not iterate over the results as len() does
+
+    # SELECT * FROM shrimp  (what len() does)
+    # SELECT COUNT(*) FROM shrimp (what count() does)
+
+    last_shrimp_born = Shrimp.objects.order_by('-birth_date').first()   # orders by birth date desc, displays the first
```
Comment (hanoskoe) [inline src/shrimps/views.py:19]: it's better to remove the "-" and use a last() function instead of first()

File: src/shrimps/views.py [hunk 2]
```diff
@@ -1,3 +1,44 @@
...
+
+
+def give_birth_to_shrimp(request):
+    """Creates a new shrimp with random name, size, weight and color and saves it to the database
+    every time the page reloads."""
+    names = ["Max", "Cooper", "Ollie", "Walter", "Apollo", "Milo", "Billy"]
+    name= choice(names)
+    size = randint(10, 40)
+    weight = randint(1, 10)
+    color = choice(Shrimp.COLOR_CHOICES)
```
Comment (hanoskoe) [inline src/shrimps/views.py:31]: your variable color here is a tuple. Please make sure you pass a string

Comment (hanoskoe) [inline src/shrimps/views.py]: your variable color here is a tuple. Please make sure you pass a string

File: src/shrimps/views.py [hunk 3]
```diff
@@ -1,3 +1,44 @@
...
+
+    new_shrimp = Shrimp(name=name, size=size, weight=weight, color=color)
```
Comment (hanoskoe) [inline src/shrimps/views.py]: you could also generate a random boolean to know if the baby shrimp is farmed

File: src/shrimps/views.py (hunk 1 above)
Comment (hanoskoe) [inline src/shrimps/views.py]: it's better to remove the "-" and use a last() function instead of first()

**Review by hanoskoe — CHANGES_REQUESTED** [review]
sorry Eli I got a bit lost in this review stuff... I hope it's understandable anyway :) good luck

File: src/shrimps/views.py [hunk 4]
```diff
@@ -1,3 +1,44 @@
...
+
+    last_shrimp_born = Shrimp.objects.order_by('-birth_date').first()   # orders by birth date desc, displays the first
+    return HttpResponse(f"""There are {num_shrimps} shrimps in our aquarium. The last shrimp born is 
+                        {last_shrimp_born}.""")
+
+
+def give_birth_to_shrimp(request):
+    """Creates a new shrimp with random name, size, weight and color and saves it to the database
+    every time the page reloads."""
+    names = ["Max", "Cooper", "Ollie", "Walter", "Apollo", "Milo", "Billy"]
```
Comment (paulwalger) [inline src/shrimps/views.py]: GERARD IS MISSING!

File: src/shrimps/views.py (hunk 1 above)
Comment (hanoskoe) [reply src/shrimps/views.py]: also make sure you handle the case num_shrimps=0

**Comment by EliNovakova** [comment]
Finally fixed @hanoskoe :)

**Comment by hanoskoe** [comment]
Looks good to me, go on with the merge!

**Pull request merged**